tests/test_causal_graph.py

```diff
@@ -538,6 +538,16 @@ def test_adjacency_consistency(self):
 
         self.assertEqual(CausalGraph.from_adjacency_matrix(*cg.to_numpy()), cg)
 
+    def test_copy(self):
+        # test copy preserves the correct class
+        ts_cg = TimeSeriesCausalGraph()
+        ts_cg.add_edge('a', 'b')
+        ts_cg.add_edge('a lag(n=1)', 'b')
+        ts_cg.add_edge('b lag(n=1)', 'b')
+
+        ts_cg_copy = ts_cg.copy()
```

Comment:
Add `self.assertIsInstance(ts_cg_copy, TimeSeriesCausalGraph))` just cause this could pass and the equality fails when something changes in the future.